ceed-vec.c

```diff
@@ -96,9 +99,14 @@ int CeedVectorSetValue(CeedVector x, CeedScalar value) {
 /// @sa CeedVectorRestoreArray()
 int CeedVectorGetArray(CeedVector x, CeedMemType mtype, CeedScalar **array) {
   int ierr;
+
+  if (x && (x->state % 2) == 1)
+    return CeedError(x->ceed, 1, "Cannot grant CeedVector array access, the access lock is already in use");
```

Comment:
I think this check should also go in `CeedVectorGetArrayRead`.

Reply:
I was thinking to allow for multiple read access. That shouldn't be problematic, should it?

Reply:
Multiple read access is fine (it doesn't modify state), but read access should not be acquired while someone is modifying.

Reply:
Ah, good point.